golem-worker-service/src/api/api_deployment.rs

```diff
@@ -182,4 +185,74 @@ impl ApiDeploymentApi {
 
         Ok(Json("API deployment deleted".to_string()))
     }
+
+    /// Undeploy a single API definition from a site
+    ///
+    /// Removes a specific API definition (by id and version) from a site without deleting the entire deployment.
+    #[oai(
+        path = "/:site/:id/:version",
+        method = "delete",
+        operation_id = "undeploy_api"
+    )]
+    async fn undeploy_api(
+        &self,
+        site: Path<String>,
+        id: Path<String>,
+        version: Path<String>,
+    ) -> Result<Json<String>, ApiEndpointError> {
+        let record = recorded_http_api_request!(
+            "undeploy_api",
+            site = site.0.clone(),
+            id = id.0.clone(),
+            version = version.0.clone()
+        );
+
+        let response = self
+            .undeploy_api_internal(site.0, id.0, version.0, &EmptyAuthCtx::default())
+            .instrument(record.span.clone())
+            .await;
+        record.result(response)
+    }
+
+    async fn undeploy_api_internal(
+        &self,
+        site: String,
+        id: String,
+        version: String,
+        auth_ctx: &EmptyAuthCtx,
+    ) -> Result<Json<String>, ApiEndpointError> {
+        let namespace = DefaultNamespace::default();
+        let api_definition_key = ApiDefinitionIdWithVersion {
+            id: ApiDefinitionId(id),
+            version: ApiVersion(version),
+        };
+
+        // Split the site string into host and subdomain parts
+        let (host, subdomain) = if let Some(idx) = site.find('.') {
+            let (subdomain, host) = site.split_at(idx);
+            (
+                host.trim_start_matches('.').to_string(),
+                Some(subdomain.to_string()),
+            )
+        } else {
+            (site, None)
+        };
+
+        let api_site = ApiSite { host, subdomain };
```

Comment:
I think `ApiSiteString` may be of your help. may be your undeploy function can take this one instead of `ApiSite`.

That to undeploy an API from a site, you really don't need to logically split domain and subdomain anymore.